app/src/main/java/su/levenetc/android/textsurface/sample/animations/complex.kt

```diff
@@ -0,0 +1,116 @@
+package su.levenetc.android.textsurface.sample.animations
+
+import android.content.res.AssetManager
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Typeface
+import su.levenetc.android.textsurface.Text
+import su.levenetc.android.textsurface.TextSurface
+import su.levenetc.android.textsurface.animations.camera.TransSurface
+import su.levenetc.android.textsurface.animations.colors.Alpha
+import su.levenetc.android.textsurface.animations.colors.ChangeColor
+import su.levenetc.android.textsurface.animations.effects.Rotate3D
+import su.levenetc.android.textsurface.animations.effects.Slide
+import su.levenetc.android.textsurface.animations.effects.reveal.Circle
+import su.levenetc.android.textsurface.animations.effects.reveal.ShapeReveal
+import su.levenetc.android.textsurface.animations.effects.reveal.SideCut
+import su.levenetc.android.textsurface.animations.generic.Delay
+import su.levenetc.android.textsurface.animations.sets.Parallel
+import su.levenetc.android.textsurface.animations.sets.Sequential
+import su.levenetc.android.textsurface.constants.Align
+import su.levenetc.android.textsurface.constants.Direction
+import su.levenetc.android.textsurface.constants.Pivot
+import su.levenetc.android.textsurface.constants.Side
+
+fun playCookieThumper(textSurface: TextSurface, assetManager: AssetManager) {
+
+    val paint = Paint().apply {
+        isAntiAlias = true
+        typeface = Typeface.createFromAsset(assetManager, "fonts/Roboto-Black.ttf")
+    }
+
+    val textDaai = Text.Builder("Daai")
```

Comment:
Seems like a using Java API from Kotlin, feels verbose. I would expect something more idiomatic to Kotlin (Data classes, type-safe builders, or just `apply` with mutable fields as you do earlier?) Maybe any reason for `set..`?

Reply:
Yeah, I agree. I actually tried to get rid of `TextBuilder`, but there two problems:
1. Making of `Position` instance has it's own requirements and there're 3 most frequent cases and they are hidden in setters:
```kotlin
val position1 = Position(align, alignText = alignText)
val position2 = Position(align)
val position3 = Position(point = PointF(px, py))
```
2. For most cases `scale` and `scale pivot` needs to be set correctly and it's done in final `build` call:
```kotlin
val text = Text(text, position, padding, paint)
text.scaleX = scale
text.scaleY = scale
text.setScalePivot(scalePivot, scalePivot)
```
So  for most cases `Builder` hides these requirements inside setters and final `build` function. For custom (and probably very rare cases with weird position and maybe inverted or negative scale) cases developer can use `Position` and `Text` constructors without `Builder`.

But it might be that these requirements can be hidden inside type-safe builder. Need to think about better api.